packages/common/src/utils/collection.ts

```diff
@@ -128,7 +128,9 @@ const checkConsentsFromValues = (publisherConsents: Consent[], consents: Record<
   if (!publisherConsents || !consents) return { customerConsents, consentsErrors };
 
   publisherConsents.forEach((consent) => {
-    if (consent.required && !consents[consent.name]) {
+    if (!consent) return;
+
+    if (!consents[consent.name] && consent) {
```

Comment:
Can an empty object occur? Or is this added to fix the typings?
The early return does the same as the following condition. I think we should use one or the other.

I'm thinking of:

```suggestion
    if (consent && !consents[consent.name]) {
```

Reply:
Oversight on my part. This should be consent.required.